test/reduce.jl

```diff
@@ -215,16 +215,16 @@ prod2(itr) = invoke(prod, Tuple{Any}, itr)
 @test maximum(Int32[1,2]) === Int32(2)
 
 A = circshift(reshape(1:24,2,3,4), (0,1,1))
-@test extrema(A,1) == reshape([(23,24),(19,20),(21,22),(5,6),(1,2),(3,4),(11,12),(7,8),(9,10),(17,18),(13,14),(15,16)],1,3,4)
-@test extrema(A,2) == reshape([(19,23),(20,24),(1,5),(2,6),(7,11),(8,12),(13,17),(14,18)],2,1,4)
-@test extrema(A,3) == reshape([(5,23),(6,24),(1,19),(2,20),(3,21),(4,22)],2,3,1)
-@test extrema(A,(1,2)) == reshape([(19,24),(1,6),(7,12),(13,18)],1,1,4)
-@test extrema(A,(1,3)) == reshape([(5,24),(1,20),(3,22)],1,3,1)
-@test extrema(A,(2,3)) == reshape([(1,23),(2,24)],2,1,1)
-@test extrema(A,(1,2,3)) == reshape([(1,24)],1,1,1)
-@test size(extrema(A,1)) == size(maximum(A,dims=1))
-@test size(extrema(A,(1,2))) == size(maximum(A,dims=(1,2)))
-@test size(extrema(A,(1,2,3))) == size(maximum(A,dims=(1,2,3)))
+@test extrema(A,dims=1) == reshape([(23,24),(19,20),(21,22),(5,6),(1,2),(3,4),(11,12),(7,8),(9,10),(17,18),(13,14),(15,16)],1,3,4)
+@test extrema(A,dims=2) == reshape([(19,23),(20,24),(1,5),(2,6),(7,11),(8,12),(13,17),(14,18)],2,1,4)
+@test extrema(A,dims=3) == reshape([(5,23),(6,24),(1,19),(2,20),(3,21),(4,22)],2,3,1)
+@test extrema(A,dims=(1,2)) == reshape([(19,24),(1,6),(7,12),(13,18)],1,1,4)
+@test extrema(A,dims=(1,3)) == reshape([(5,24),(1,20),(3,22)],1,3,1)
+@test extrema(A,dims=(2,3)) == reshape([(1,23),(2,24)],2,1,1)
+@test extrema(A,dims=(1,2,3)) == reshape([(1,24)],1,1,1)
+@test size(extrema(A,dims=1)) == size(maximum(A,dims=1))
+@test size(extrema(A,dims=(1,2))) == size(maximum(A,dims=(1,2)))
+@test size(extrema(A,dims=(1,2,3))) == size(maximum(A,dims=(1,2,3)))
 
```

Comment:
Would be nice to test `extrema(A, dims=:)` as well.

Reply:
There are other tests for `extrema(A)`, which is that method.

Reply:
Indeed the correctness of `extrema(A)` is already tested (I guess for `extrema(A)` 1-argument version from `reduce.jl` gets called).
I've meant testing that `extrema(A, dims=:)` is dispatched properly.
Actually, wouldn't it be better to remove the default `:` value for `dims`?
Its return value is `Tuple{T, T}`, whereas for the other `dims` values it's `Array{Tuple{T,T}, N}`.

Reply:
Other functions with the `dims` keyword work this way too. Also the `:` is kind of unofficial, and mostly a hack needed to make the keyword argument work. But it makes enough sense that it's fine if people use it.